File: Algorithms/Implementation/Beautiful Triplets.py
import os

#
# Complete the 'beautifulTriplets' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER d
#  2. INTEGER_ARRAY arr
#

def beautifulTriplets(d, arr):
    # Write your code here
    count = 0
    for i, val_i in enumerate(arr):
        for j, val_j in enumerate(arr[i+1:], start=i+1):
            if val_j - val_i == d:
                for k, val_k in enumerate(arr[j+1:], start=j+1):
                    if val_k - val_j == d:
                        count += 1
    return count


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()
    n = int(first_multiple_input[0])
    d = int(first_multiple_input[1])

    arr = list(map(int, input().rstrip().split()))
    result = beautifulTriplets(d, arr)

    fptr.write(str(result) + '\n')
    fptr.close()

File: Algorithms/Implementation/Beautiful Triplets.php
<?php

/*
 * Complete the 'beautifulTriplets' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER d
 *  2. INTEGER_ARRAY arr
 */

function beautifulTriplets($d, $arr) {
    # Write your code here
    $count = 0;
    for($i = 0; $i < count($arr) - 2; $i++) {
        for($j = $i + 1; $j < count($arr) - 1; $j++) {
            if($arr[$j] - $arr[$i] == $d) {
                for($k = $j + 1; $k < count($arr); $k++) {
                    if($arr[$k] - $arr[$j] == $d) {
                        $count++;
                    }
                }
            }
        }
    }
    return $count;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$first_multiple_input = explode(' ', rtrim(fgets(STDIN)));
$n = intval($first_multiple_input[0]);
$d = intval($first_multiple_input[1]);
$arr_temp = rtrim(fgets(STDIN));

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));
$result = beautifulTriplets($d, $arr);

fwrite($fptr, $result . "\n");
fclose($fptr);
